NetSuite 11.0 New (#1335)
mulesoft/docs-connectors: netsuite/11.0/modules/ROOT/pages/netsuite-reference.adoc

= NetSuite Connector 11.0 Reference

Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Select]

> Anypoint Connector for NetSuite synchronizes data and automates business processes between NetSuite and third party applications, either on-premise or in the cloud. The connector makes use of the SuiteTalk WSDL to provide SOAP based integration to generate NetSuite business objects, make use of different authentication levels, and support error handling.

Reviewer: ```suggestion
Anypoint Connector for NetSuite synchronizes data and automates business processes between NetSuite and third party applications, either on-premises or in the cloud. The connector makes use of the SuiteTalk WSDL to provide SOAP based integration to generate NetSuite business objects, make use of different authentication levels, and support error handling.
```

| Ignore read-only fields a| Boolean |  This preference affects the way the system behaves when you mistakenly submit a value for a read-only field. If you set this preference to `true`, the system ignores these mistakes. |  false |
| Run server SuiteScript and trigger workflows a| Boolean a|  Controls SuiteScript and trigger workflows per request:

* If you do not include this preference in a request, the company preference set on the Web Services Preferences page is used.
* If you include this preference in the request, the connector overrides the company preference set in the UI. |  false |
| Customizations Separator a| String |  Separator used to generate the keys required to provided better support for custom fields and records. If the script ID of the customization that defines a custom field or record contains the separator default character, set up a new ID and regenerate the metadata. |  __ |
|===

==== Connection Types

The connection types to provide to this configuration.

[[Config_TokenBasedAuthentication]]

===== Token-Based Authentication

====== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required

> | Maximum Connections a| Number |  The maximum amount of connections that can be active to execute the HTTP requests. The default value, `-1`, does not limit the amount of connections to the server. |  -1 |

Reviewer: ```suggestion
| Maximum Connections a| Number |  The maximum number of connections that can be active to execute the HTTP requests. The default value, `-1`, which does not limit the number of connections to the server. |  -1 |
```

| Read Timeout a| Number |  Specifies the amount of time that the client waits for a response before it times out. Use `0` to specify that the client should wait indefinitely. |  60000 |
| Read Timeout Unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  The time unit for Read Timeout. |  MILLISECONDS |
| Connection Timeout a| Number |  Specifies the amount of time that the client attempts to establish a connection before it times out. Use `0` to specify that the client continue to attempt to open a connection indefinitely. |  60000 |
| Connection Timeout Unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS
** DAYS |  The time unit for Connection Timeout. |  MILLISECONDS |

> | Base Netsuite Address a| String |  The base address to use to connect to NetSuite. The default value is the main NetSuite site. This is used to automatically obtain the endpoint for the service based on the credentials configured, and will need no change. |  https://webservices.netsuite.com |

Reviewer: ```suggestion
| Base Netsuite Address a| String |  The base address to use to connect to NetSuite. The default value is the main NetSuite site. This is used to automatically obtain the endpoint for the service based on the credentials configured, and require no change. |  https://webservices.netsuite.com |
```

> | SOAP Port a| String |  The SOAP Port path to use to connect to NetSuite, this is relative to the base address. The default value corresponds to the port obtained from the WSDL used in the version of the connector |  | x

Reviewer: ```suggestion
| SOAP Port a| String |  The SOAP Port path to use to connect to NetSuite, this is relative to the base address. The default value corresponds to the port obtained from the WSDL used in the version of the connector. |  | x
```

| Reconnection Strategy a| * <<Reconnect>>
* <<ReconnectForever>> |  A retry strategy in case of connectivity errors |  |
|===

=== Output

[%autowidth.spread]
|===
|Type |Binary
| Attributes Type a| <<NetsuiteSoapAttributes>>
|===

=== For Configurations

* <<Config>>

=== Throws

* NETSUITE:CONNECTIVITY
* NETSUITE:INVALID_VERSION
* NETSUITE:NETSUITE_ERROR
* NETSUITE:NETSUITE_SOAP_FAULT
* NETSUITE:RETRY_EXHAUSTED
* NETSUITE:SESSION_TIMED_OUT
* NETSUITE:USER_ERROR

[[AddList]]
== Add List
`<netsuite:add-list>`

Creates one or more new records.

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Type a| String |  The record type for the operation. |  |
| Message a| Any |  The add list request to complete. |  #[payload] |
| Config Ref a| ConfigurationProvider |  The name of the configuration to use to execute this component |  | x
| Streaming Strategy a| * <<RepeatableInMemoryStream>>
* <<RepeatableFileStoreStream>>
* non-repeatable-stream |  Configures how Mule processes streams. Repeatable streams are the default behavior. |  |

> | Target Variable a| String |  The name of a variable to store the operation's output. |  |

Reviewer: ```suggestion
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
```

Reviewer: It's up to you whether or not you want to S&R this.
